### PR TITLE
[Schema Registry] Generating from TypeSpec for JSON GA

### PR DESCRIPTION
**Pull request by m-redding**
This PR transitions the Schema Registry SDK to be generated from TypeSpec instead of Swagger.

There are a few outstanding issues in the generated code related to the internal implementation.
- Header deserialization is no longer generated, so I have to manually extract the headers. This is also an issue in Java. There are some design/feature discussions happening but no commitment on this being added at this time.
- The content type class is no longer generated (even though it is in Python). This is a bug and will hopefully be fixed.

There is no ETA for either of these issues, but I can work around them in the handwritten code, so I don't think we should postpone transitioning to TypeSpec because of this.

### CODE REVIEW

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/src/Generated/Models/ContentType.cs [hunk 1]
```diff
@@ -1,36 +0,0 @@
-// Copyright (c) Microsoft Corporation. All rights reserved.
-// Licensed under the MIT License.
-
-// <auto-generated/>
-
-#nullable disable
-
-using System;
```
Comment (m-redding) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/src/Generated/Models/ContentType.cs:8]: This is a bug in the emitter, this class should not be deleted and wasn't previously: https://github.com/Azure/autorest.csharp/issues/4897

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs [hunk 2]
```diff
@@ -34,6 +34,7 @@ public partial class SchemaRegistryClient
         public SchemaRegistryClient(string fullyQualifiedNamespace, Azure.Core.TokenCredential credential) { }
         public SchemaRegistryClient(string fullyQualifiedNamespace, Azure.Core.TokenCredential credential, Azure.Data.SchemaRegistry.SchemaRegistryClientOptions options) { }
         public string FullyQualifiedNamespace { get { throw null; } }
+        public virtual Azure.Core.Pipeline.HttpPipeline Pipeline { get { throw null; } }
```
Comment (m-redding) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:37]: I can try to hide this, but I think it's just part of the DPG API surface.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs [hunk 3]
```diff
@@ -45,12 +46,12 @@ public partial class SchemaRegistryClient
...
     public partial class SchemaRegistryClientOptions : Azure.Core.ClientOptions
     {
-        public SchemaRegistryClientOptions(Azure.Data.SchemaRegistry.SchemaRegistryClientOptions.ServiceVersion version = Azure.Data.SchemaRegistry.SchemaRegistryClientOptions.ServiceVersion.V2023_07) { }
+        public SchemaRegistryClientOptions(Azure.Data.SchemaRegistry.SchemaRegistryClientOptions.ServiceVersion version = Azure.Data.SchemaRegistry.SchemaRegistryClientOptions.ServiceVersion.V2023_07_01) { }
         public enum ServiceVersion
         {
             V2021_10 = 1,
             V2022_10 = 2,
-            V2023_07 = 3,
+            V2023_07_01 = 3,
```
Comment (m-redding) [inline sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:54]: This hasn't been released yet.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs (hunk 2 above)
Comment (jsquire) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:37]: It shows up in most libraries when you search the repo - may as well just let it stick around and join the crowd.

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs (hunk 3 above)
Comment (jsquire) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:54]: Is this the protobuf one or just one that we meant to release for JSON?

**Review by jsquire — APPROVED** [review]

File: sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs (hunk 3 above)
Comment (m-redding) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:54]: @jsquire it is the Protobuf one

Comment (m-redding) [reply sdk/schemaregistry/Azure.Data.SchemaRegistry/api/Azure.Data.SchemaRegistry.netstandard2.0.cs:54]: 2022_10 is the JSON/Custom one and has been released but only in beta

**Pull request merged**